execution_engine_testing/test_support/src/chainspec_config.rs

```diff
@@ -39,9 +39,20 @@ pub enum Error {
         error: io::Error,
     },
     FailedToParseChainspec(toml::de::Error),
-    FailedToCreateExecConfig,
-    FailedToParseLockedFundsPeriod,
-    FailedToCreateGenesisRequest,
+}
+
+fn vesting_schedule_period<'de, D: Deserializer<'de>>(
```

Comment:
Again, let's not do this to validate config settings.  This should be changed to match how we do it in the "real" config, ideally sharing code if possible.

Reply:
Ticketed the refactor to share most code (# to be announced). For now, refactored just enough to always call the new method `fn is_valid()` before returning a valid `ChainspecConfig` instance.